### DISCUSSION TITLE
Mac Quick Look

### DISCUSSION THREAD

**Discussion by jnathanh** [saulpw/visidata#1734] [Ideas]
Hi, I'm a new VisiData user and absolutely love it so far. It solves so many problems I've been looking for a solution for, so thank you.

My Idea: enable VisiData as a Mac Quick Look provider

One of the things I love most about VisiData is that it will let me quickly inspect most files because of it's extensive support of different file formats. macOS's Quick Look feature is similarly nice because it is a very convenient way to preview content from Finder, though it doesn't support many of the filetypes that VisiData does. Seems like a good pairing for VisiData's strengths and would be an easy entrypoint to VisiData when you're not already in the terminal.

**Comment by saulpw** [comment]
Hey @jnathanh, thanks for the suggestion!  I took a quick look (no pun intended) at [the Quick Look Architecture](https://developer.apple.com/library/archive/documentation/UserExperience/Conceptual/Quicklook_Programming_Guide/Articles/QLArchitecture.html), and it seems like it should be doable. I'm thinking that the preview app would only load some ~100 rows and then save it as html in the right place for QL to render.  We'd need to implement the feature to stop loading after some number of rows, but that seems reasonable.

As I'm not a Mac user myself, I'll leave it to someone who knows the platform better and can test it sufficiently.  I'd be happy to include it if it works reasonably well!

**Comment by jnathanh** [comment]
Thanks for the feedback @saulpw. I agree that the displayed content would only need to be a subset of the actual document. My main use-case for the feature is to peek at the file to identify it or understand what is in it.

I mostly wanted to share the idea, and do not have time to work on it right now. I may circle back in the future to try my hand at it though.
